### ISSUE TITLE
Keyboard pops up after selection is made on mobile devices

### ISSUE THREAD

**Issue by j-w-miller** [angular-ui/ui-select#818]
Banging my head on this for a while, hoping somebody can point me in the right direction.  On iOS the keyboard is displayed momentarily, however on android it persists when viewed with Firefox or Chrome and is not removed until it looses focus.  Here's the code, not much to it:

```html
<ui-select ng-model="$parent.source_type" theme="bootstrap" title="Choose source type" search-enabled="false">
  <ui-select-match  placeholder="Select a source type ...">{{$select.selected.name}}</ui-select-match>
  <ui-select-choices  repeat="type in source_types">{{type.name}}</ui-select-choices>
</ui-select>
```

The scenario is:
* Web page is displayed.
* User touches ui-select element and drop-down opens up.
* User selects (touches) desired element from drop-down, drop-down is closed displaying the user selected element.
* Keyboard is displayed <<< undesirable event

Am I missing something?
Any ideas on how I could prevent this from occurring?
Is this a bug or feature? `<grin>`

Thanks for looking,
John

**Comment by leoncc** [comment]
Had the same issue with the keyboard on Android with Chrome, popped up twice when the control got focus and after an item has been selected. Here is a quick and dirty fix for the two unwanted keyboard popups (in reverse order).

In select.js, the ctrl.close....., comment out:
   //$scope.$broadcast('uis:close', skipFocusser);

The focusser seems to be a dummy element that receives focus and it looks like it causes the keyboard to pop up, when you select an item and the dropdown closes.

Where it does the #templateCache thing right at the end of select.js, last line (assuming you use selectize):

$templateCache.put("selectize/select.tpl.html","&lt;div class=\"ui-select-container selectize-control single\" ng-class=\"{\'open\': $select.open}\">&lt;div class=\"selectize-input\" ng-class=\"{\'focus\': $select.open, \'disabled\': $select.disabled, \'selectize-focus\' : $select.focus}\" ng-click=\"$select.activate()\">&lt;div class=\"ui-select-match\">&lt;/div>&lt;input ng-readonly=\"!$select.searchEnabled\"  type=\"text\" autocomplete=\"off\" tabindex=\"-1\" class=\"ui-select-search ui-select-toggle\" ng-click=\"$select.toggle($event)\" placeholder=\"{{$select.placeholder}}\" ng-model=\"$select.search\" ng-hide=\"($select.selected && !$select.open)\" ng-disabled=\"$select.disabled\" aria-label=\"{{ $select.baseTitle }}\">&lt;/div>&lt;div class=\"ui-select-choices\">&lt;/div>&lt;/div>");}]);

So moving searchEnabled to ng-readonly (instead of ng-hide) lets you keep the placeholder text, does not cause the first keyboard popup and does not cause the height shrinkage, which I also experienced.

Then also where you use the control in your view in the markup: ...... search-enabled="false".

I did not test this thoroughly for side effects.

**Comment by NetanelBasal** [comment]
i have this problem too, you resolve this in elegant way?

**Comment by j-w-miller** [comment]
No, I reverted back to using html <select>.  I thought leoncc's proposal was interesting, but was too near release of our software to take the risk.

**Comment by NetanelBasal** [comment]
Ok thanks.

**Comment by greenwebdev** [comment]
+1 for this, would love a fix.

**Comment by K1N5L4Y3R** [comment]
+1

**Comment by bereziuk** [comment]
+1

**Comment by zenato** [comment]
+1 i have this problem too. How to solve this problem?

**Comment by ebergama** [comment]
+1 here!

**Comment by sergei1152** [comment]
Tried using jquery blur events on the right elements after a selection with no luck. +1 for a fix.

**Comment by RuneKobberoe** [comment]
+1

**Comment by mgleed** [comment]
+1

**Comment by cash1981** [comment]
+1

**Comment by wesleycho** [comment]
This is also related to the broader discussion on event handlers with the library.

**Comment by karas111** [comment]
+1

**Comment by VictorAtPL** [comment]
+1

**Comment by rob-balfre** [comment]
Careful removing this line:
`$scope.$broadcast('uis:close', skipFocusser); `

Better off using the skip-focusser attribute.
See docs https://github.com/angular-ui/ui-select/wiki/ui-select

**Comment by francescjimenez** [comment]
+1

**Comment by vishnutk** [comment]
+1

**Comment by osamaurrehman** [comment]
I simply did by writing some css to prevent the keyboard to popup on mobile devices. 
.ui-select-container .ui-select-search, .ui-select-container .ui-select-focusser  {
        visibility: hidden;
}

You can do this for mobiles specifically by detecting touch devices and adding a class to your body for mobile, like
.classForMobile .ui-select-container .ui-select-search, .classForMobile  .ui-select-container .ui-select-focusser  {
        visibility: hidden;
}

**Comment by sushruth** [comment]
@osamaurrehman That saved my life today! thanks.

**Comment by ashwin713** [comment]
@leoncc Thanks a lot :)

**Comment by thienedits** [comment]
I was able to create a directive

```
(function () {
  'use strict';

  angular
    .module('app.directives', [])
    .directive('readOnly', readOnly);

  function readOnly() {
    return function (scope, element, attrs) {
      element.find("input").attr('readonly', 'readonly');
    };
  }

})();
```

then in the template add the read-only attribute
```
<ui-select multiple ng-model="group.selectedCategories" theme="bootstrap" ng-disabled="false" sortable="true" close-on-select="false" remove-selected="false" read-only>
```

**Comment by tarastmb** [comment]
part 1. for project requirements on "select.tpl.html" we already have:
`ng-show="$select.searchEnabled && $select.open">`
so the input hides when `search-enabled` is false

part 2. to solve the other keyboard placed on this focusser:
`<input ng-disabled="$select.disabled" class="ui-select-focusser ui-select-offscreen ng-scope" type="text" id="focusser-0" aria-label="Select box focus" aria-haspopup="true" role="button" disabled=""></div>`

I did a CSS fix for mobile resolutions and it works fine:

```
@media (max-width: @screen-xs-max) {
  .ui-select-offscreen {
    display: none;
  }
}
```

**Comment by MatejQ** [comment]
+2

**Comment by timkishkin** [comment]
I've found the following workaround useful: I added `skip-focusser="isMobileDevice"` attribute, where isMobileDevice is my bool scope variable set during mobile device detection. Thus nothing is changed on desktop.
